File: bzk/bzkflow/src/main/java/net/bzk/flow/model/BzkObj.java
package net.bzk.flow.model;

import org.apache.commons.lang3.RandomStringUtils;

import lombok.Data;
import net.bzk.infrastructure.convert.OType;

@Data
public class BzkObj implements OType {

	private String clazz;
	private String uid;

	public BzkObj() {
		clazz = this.getClass().getName();
		uid = RandomStringUtils.randomAlphanumeric(32);
	}

}

File: flow-editor/src/app/model/bzk-obj.ts
import { PropInfo, PropType } from '../utils/prop-utils';

export class OType {
  @PropInfo({
    title: 'clazz',
    type: PropType.Label
  })
  public clazz = '';
}

export class BzkObj extends OType {


  @PropInfo({
    title: 'uid',
    type: PropType.Label
  })
  public uid = '';


}
